### PR TITLE
🍰 feature/backend api

### PR DESCRIPTION
**Pull request by Olderestin**
## What changed?     
Create API endpoint for:
- user model with JWT auth
- auction, bid and auction image model
Use docker compose container for using app.

## Checklist before requesting a review     
- [x] I have performed a self-review of my code
- [x] Code is functional on my machine(Windows 10/Debian GNU/Linux 11 (bullseye))

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,77 @@
...
+   ```bash
+   git clone https://github.com/Olderestin/web_project_auction.git
+
+2. Navigate to the project directory:
+   ```bash
+   cd web_project_auction
+
+3. Change branch:
+   ```bash
+   git checkout feature/backend_api
```
Comment (memhard) [inline README.md]: we doesn't that step!

File: auction/serializers.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        auction = validate_data["auction"]
+        bid_value = validate_data["bid"]
+
+        if not auction.bid:
+            if auction.start_bid <= bid_value:
+                bid = self.create_bid(user, validate_data, auction)
+            else:
+                raise serializers.ValidationError(
+                    "Bid value must be at least equal to the start bid."
+                )
```
Comment (memhard) [inline auction/serializers.py:58]: there doesn't need else construction

File: auction/serializers.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+                "Bid value must be higher than the current bid plus the bid step."
+            )
+
+        return bid
+
+    def create_bid(self, user, validate_data: Dict[str, Any], auction: Auction) -> Bid:
+        bid = Bid.objects.create(user=user, **validate_data)
+        auction.bid = bid.bid
+        auction.save()
+        return bid
```
Comment (memhard) [inline auction/serializers.py]: that method must be higher than create method, because you use this method in create func

File: .env.example [hunk 4]
```diff
@@ -0,0 +1,12 @@
+SECRET_KEY = ""
+
+ALLOWED_HOSTS_1 = "127.0.0.1"
+
+ALLOWED_ORIGINS_1 = "http://localhost:8000"
+ALLOWED_ORIGINS_2 = "http://localhost:5173"
```
Comment (memhard) [inline .env.example]: use one variable, example: "http://localhost:8000,http://localhost:5173"

File: config.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+class Settings(BaseSettings):
+    """
+    Class for storing app settings.
+    """
+
+    SECRET_KEY: str
+
+    ALLOWED_HOSTS_1: str
+    ALLOWED_ORIGINS_1: str
+    ALLOWED_ORIGINS_2: str
```
Comment (memhard) [inline config.py]: same

**Review by memhard — APPROVED** [review]
LGTM 👍 can be approved, but still need some changes

**Review by Olderestin — COMMENTED** [review]
Changes done.

**Review by memhard — APPROVED** [review]
LGTM 👍 can approve

**Pull request merged**